### PR TITLE
RealmResultsChanges.isCleared 

### LINKED ISSUE #1189: Add test for RealmResultsChanges.isCleared
Convert a RealmList to RealmResults using asResults and then clear the list.

### PR DESCRIPTION
**Pull request by desistefanova**
Deprecate `isCleared` for `RealmResults`.
Fixes #1189
Related to https://github.com/realm/realm-dart/issues/1278

### CODE REVIEW

**Review by nirinchev — CHANGES_REQUESTED** [review]
This doesn't seem correct to me. You can convert a list to results and subscribe for notifications. I would expect that it to emit `isCleared` correctly.

**Comment by desistefanova** [comment]
> This doesn't seem correct to me. You can convert a list to results and subscribe for notifications. I would expect that to emit `isCleared` correctly when the underlying list is cleared.

Ok, I will revert these changes and will add a new test. Thank you for noticing this @nirinchev!

**Review by nielsenko — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 * None
 
 ### Fixed
-* None
+* `RealmResultsChanges.isCleared` is deprecated. Its value was not set correctly. It is fixed to return true if the results `isEmpty` ([#1265](https://github.com/realm/realm-dart/pull/1265)). We are considering changing this API in future releases ([#1278](https://github.com/realm/realm-dart/issues/1278)).
```
Comment (blagoev) [inline CHANGELOG.md:7]:
```suggestion
* Fixed `RealmResultsChanges.isCleared` which was never set. It now returns `true` if the results collection is empty in the notification callback. This field is also marked as `deprecated` now and will be removed in future. Use `RealmResultsChanges.results.isEmpty` instead.([#1265](https://github.com/realm/realm-dart/pull/1265)). ([#1278](https://github.com/realm/realm-dart/issues/1278)).
```

**Pull request merged**